Sources/SwiftKueryMySQL/MySQLConnection.swift

```diff
@@ -130,6 +133,7 @@ public class MySQLConnection: Connection {
 
         let connectionGenerator: () -> Connection? = {
             let connection = self.init(host: host, user: user, password: password, database: database, port: port, unixSocket: unixSocket, clientFlag: clientFlag, characterSet: characterSet, reconnect: reconnect)
```

Comment:
Would it be better to pass the timeout through as an `init()` parameter instead?

Reply:
I have some concerns about compatibility for people who already implemented the lib.
_ I can simply modify the current (designated) `required init()` but if anyone subclassed `MySQLConnection`, they would have compile time error when updating their project.
_ I can make a new designated initializer (as required) so the current one become convenience? People will also have compile time error ?
_ I can make a new convenience initializer with timeout parameter but a convenience initializer would have more parameters than the designated? (Sounds strange to me)

That's why I thought not changing the current `init()` preferable.

Again, I'm not an expert so tell me what sounds the best for you, and I will try.

Reply:
These are good points, let's leave this as it is then :)